qamomile/core/post_process/local_search.py

```diff
@@ -0,0 +1,119 @@
+"""
+qamomile/core/post_process.py
+
+This module provides functionality for Local Search Algorithm. Local Search Algorithm is one of the simplest 
+heuristic algorithms for the optimization problem. Let's take the Ising Hamiltonian as objective
+function to be minimized. 
+
+    .. math::
+        \sum_{ij} J_{ij} z_i z_j + \sum_i h_i z_i, ~\text{s.t.}~z_i \in \{-1, 1\}
+            
+Key Features:
+- Implementation of first- and best-improvement local search algorithms
+- Construction of cost Hamiltonians for the Ising model
+- Conversion of Ising interactions from dictionary format to matrix form
+"""
+
+from qamomile.core.ising_qubo import IsingModel
+from qamomile.core.converters.converter import QuantumConverter
+from qamomile.core.bitssample import BitsSample, BitsSampleSet
+import numpy as np
+import jijmodeling as jm
+from typing import Callable
+
+class IsingMatrix:
+    def __init__(self, quad: np.array = None, linear: np.array = None):
```

Comment:
np.array is not type. You have to use `np.ndarray`.
<img width="567" alt="image" src="https://github.com/user-attachments/assets/97b8dd83-99c4-42b0-a273-5184fa24f7ef">


If you are using vscode, installing the python extension will highlight these kinds of basic mistakes in red. Use the editor's functions to reduce these kinds of writing mistakes.

Python will still work even if the type hints are incorrect, so you need to use the editor's functions for these parts.

Also, in terms of type, None is the initial value, so the type is not np.ndarray in the first place. It needs to be wrapped in Optional.

```python
import typing as typ

...(quad: typ.Optional[np.ndarray], ...
```

It needs to be like this.